src/services/favourites.ts

```diff
@@ -0,0 +1,78 @@
+import toQueryString from "../lib/toQueryString"
+import { Paginated, SearchListing, WithValidationError } from "../index"
+import {
+  ApiCallOptions,
+  deletePath,
+  fetchPath,
+  handleValidationError,
+  postData,
+} from "../base"
+
+export const fetchFavourites = async ({
+  offset,
+  page,
+  size,
+  options = {},
+}: {
+  offset?: number
+  page?: number
+  size?: number
+  options?: ApiCallOptions
+} = {}): Promise<Paginated<SearchListing>> => {
+  const query = toQueryString({ offset, page, size })
+  return fetchPath({
+    path: `users/me/favorite-listings${query ? "?" + query : ""}`,
+    options: { isAuthorizedRequest: true, ...options },
+  })
+}
+
+export const saveFavourite = async ({
+  listingId,
+  options = {},
+}: {
+  listingId: number
+  options?: ApiCallOptions
+}): Promise<Response> => {
+  return postData({
```

Comment:
just to confirm, is there no validation error for this one if for example the listing is no longer active or something like that?

Reply:
Technically there is but the way we use the APIs we guarantee that the listing exists (since you favourite it from the SRP or PDP where it's displayed)